wordpress-mobile/WordPress-Android
# Improvements: Dashboard UI

## Fixes
#19772


## Description
This PR makes the following changes 
1.  Removes the site info header from my site fragment - A `SiteInfoHeaderCardViewHolder` is added to render the site header instead of my site fragment. This will improve the rendering time and will also help us when converting the views to compose 
2. Fixes the background color in the compose cards 

| Before  | After   |
|---|---|
| <img width="660" alt="Screenshot 2023-12-07 at 6 59 17 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/14186d0c-b291-4ddf-a06d-ff3b14adbaf5"> | <img width="638" alt="Screenshot 2023-12-12 at 4 25 32 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/fd889a32-84e0-493e-b377-3c9b50176b4c"> |  


 - Blaze Campaign
 - Bloganuary Nudge Card
 - Personalize card 
 - Domain dashboard card 

3. Adds a haptic feedback to the bottom nav 

## To Test:
1. Login to the Jetpack app 
2. Go to the dashboard 
3. Verify that the dashboard is rendered as before
5. Verify that there are no background in compose cards 
6. Verify that on switching between the bottom nav items(my site, reader, notifications, me) , there is a haptic feedback 


## Regression Notes

1. Potential unintended areas of impact
N  

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)


PR submission checklist:

UI Changes testing checklist:
